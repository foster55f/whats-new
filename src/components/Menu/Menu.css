.menu {
    margin-top: 5%;
} 

/* .menu button {

 }  */