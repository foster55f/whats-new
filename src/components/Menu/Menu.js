import React from 'react';
import './Menu.css'

const Menu = (props) => {
    const filterClick = (event) => {
    props.selectNews(event.target.className)
    }

    return (
        <div className="menu">
            <button onClick={filterClick} className='local'>Local News </button>
            <button onClick={filterClick} className='entertainment'>Entertainment </button>
            <button onClick={filterClick} className='science'>Science </button>
            <button onClick={filterClick} className='health'>Health </button>
            <button onClick={filterClick} className='technology'>Technology </button>          
        </div>
    )
}
export default Menu;