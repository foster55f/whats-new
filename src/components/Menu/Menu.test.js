import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';
import { isTSAnyKeyword } from '@babel/types';

describe('Menu', () => {

    it('should match the snapshot with all the data pulled in correctly', () => {
        const wrapper = shallow(<Menu selectNews='entertainment' />)
        expect(wrapper).toMatchSnapshot();
    });  
})


    // it('should call the filterClick prop method with className when clicked', () => {
    //     // Setup
    //     const mockEvent = { target: { value: 'Sweaters for pugs.'} };
    //     const filterClickMock = jest.fn();
    //     const wrapper = shallow(
    //       <Menu
    //         filterClick={mockEvent}
    //       />
    //     );
      
    //     // Execution
    //     wrapper.find('button').at(1).simulate('click');
      
    //     // Expectation
    //     expect(wrapper.state('target')).toEqual('className');
    //   });
