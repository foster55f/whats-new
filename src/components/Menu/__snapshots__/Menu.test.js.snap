// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot with all the data pulled in correctly 1`] = `
<div
  className="menu"
>
  <button
    className="local"
    onClick={[Function]}
  >
    Local News 
  </button>
  <button
    className="entertainment"
    onClick={[Function]}
  >
    Entertainment 
  </button>
  <button
    className="science"
    onClick={[Function]}
  >
    Science 
  </button>
  <button
    className="health"
    onClick={[Function]}
  >
    Health 
  </button>
  <button
    className="technology"
    onClick={[Function]}
  >
    Technology 
  </button>
</div>
`;
